docs/flow-cache-design-doc.md

```diff
@@ -0,0 +1,178 @@
+# Flow Cache Design Doc
+
```

Comment:
I believe that this proposal would not work for the device auth grant. Here, the oauth2 flow will be executed on two devices where the non-browser device initiates the flow, and the browser completes the login/consent dance, and finally the non-browser device finalizes the flow by polling and/or getting a verified.

Here, the cookie is not shared between devices. We might find another way to get this working though.